### ISSUE TITLE
`no-unused-expressions` behavior does not match docs for `allowShortCircuit: true`

### ISSUE THREAD

**Issue by IanVS** [eslint/eslint#3675]
According to the docs, this should not cause warnings:

```js
/*eslint no-unused-expressions: [2, { allowShortCircuit: true }]*/

a && b()
```

But, it gives:

```
3:1  error  Expected an assignment or function call and instead saw an expression  no-unused-expression
```

The second example given, (`a() || (b = c)`) behaves correctly.

**Comment by IanVS** [comment]
I don't see how this would be related to short-circuiting, so I'm guessing it should be removed from the docs.

**Comment by mysticatea** [comment]
I will try to fix this.
